### ISSUE TITLE
Help users see send transactions are pending/working as intended

### ISSUE THREAD

**Issue by 314159265359879** [blockstack/blockstack-app#2129]
I notice that new users are inclined to send a new transaction, or the same transaction again when the previous is not mined within a seconds or minute(s) or so. I know part of the problem is ignorance: users do not always realize block times for Stacks' main chain are 10-20 minutes rather than seconds or minutes.

Is there anything that can be done in the UI to help convey this:
1. On the confirmation tab show some expandable “i” with: Confirmation times between 10-30 minutes on Stacks are quiet normal.
2. Or show a status bar on the main screen of the wallet, with the number of pending transactions and some “loading” animation.  

3. If the same transactions is being send again (all arguments equal, but the nonce+1): show a prompt detailing 1. and asking to confirm sending the same transaction again.
This prompt should also have an option to not show the prompt again. (in case someone really intends to send the same transaction several times: example: `mint-10` on a popular NFT-contract). 

This issues also happens to add to mempool “congestion” with transactions that cannot be mined. When users are spending their STX on the first transaction and they can’t actually pay for the fees on following transactions. 
Together with issues causing unmineable transactions: #2096, #1991 it can contribute to even more congestion and eventually influence the API performance?

**Comment by 314159265359879** [comment]
In time of congestion you will see users posting this with "can you help? I have increased the fee but it is still pending"

![image](https://user-images.githubusercontent.com/33360391/162901815-0175948a-c1b0-4d71-8315-6f6ed8e9b7ae.png)

3. above would help a lot against this I think.

**Comment by Hero-Gamer** [comment]
I've commented this on another GH issue: https://github.com/hirosystems/explorer/issues/622
Which discusses the Explorer and Wallet.

I agree until there is a more automated solution, before that occur, if we have to manually diagnose and allow user to have the option to increase tx fee, being able to select the right one and increase to the right amount is super helpful and will cut down the support time. 
In summary I'd vouch for having **Fee** and **Nonce** displayed both in the **Wallet** & **Explorer.** 
@wileyj @whoabuddy

**Comment by whoabuddy** [comment]
I'll +1 the request to show fees and nonce, even if it's an "advanced setting" one could toggle on.

Otherwise it can take quite a few explorer pages and flipping back and forth to confirm things, especially when trying to support someone else.

**Comment by wileyj** [comment]
I'm more of the opinion that this could (should?) be part of the explorer vs the wallet itself. 

basically, a modal or something that can determine if the previous nonce is -1 of the current tx, and if the previous tx has been pending for x blocks, maybe display some warning message like "this tx is not likely to be confirmed". 

i don't think his  should be part of the wallet though, unless it's information that can be easily determined from the API data natively.

**Comment by whoabuddy** [comment]
@wileyj I think nonce is a good thing to have on the explorer with some additional logic (current vs pending, missing nonces, etc), but it would be nice to have fees visible in the wallet per transaction because that's the place where you decide to RBF.

It shows the original fee for the individual transaction when you click "Increase fee", but if you want to compare the transactions it's back to opening up tabs in the explorer. It also helps with seeing how much you can increase the fee based on your balance and other pending tx.

**Comment by Hero-Gamer** [comment]
I would echo @whoabuddy from the perspective of either users helping themselves, or us supporting users in addition, any extra pieces of info I don't have to keep clicking into look each tx individually in the Explorer, saves time.  

> There has been a recent pull thing (sorry not techny here..) about explorer display:
https://github.com/hirosystems/explorer/pull/624
I came in a bit late to mention the points I think was little too late. Maybe our current thread can serve as the next update?

> Werner (he also solves a lot of wallet issues) also recently filed this feature request: https://github.com/hirosystems/stacks-wallet-web/issues/2365
cc: @314159265359879

So I think wallet and explorer should be in the conversation together. 

Ideally for me: 
1. First have all info from Hiro wallet itself, so we (or the user) don't need to even go into Explorer unless some other more advanced issues. 
2. If cannot identify issue from wallet itself, then user/or support person can venture into the Explorer. 

This will in the medium term allow users potentially be trained easier to Identify and solve problem by themselves easier. 
I hope I'm making some sense sorry if I'm not! 

P.s. I 'm seeing a lot of Nonce gap recently. Perfection will be maybe from the wallet itself if Nonce gap does exist, the wallet can warn the user somehow. but I know I'm shooting for the moon here :)

**Comment by wileyj** [comment]
I see the points both @Hero-Gamer and @whoabuddy are making - but let's wait for someone like @markmhx to chime in here before we get ahead of ourselves with too many ideas/options.

**Comment by whoabuddy** [comment]
The other day I went through a scenario that made me think of this issue. I performed a few actions across different projects, including:

- 1 tx: claiming a CityPacks NFT
- 2 tx: buying NFTs from 2 different marketplaces
-  2tx: freezing Stacks Pops NFTs

Before submitting the last two transactions I was trying to determine the information below as there was a slow 60min Bitcoin block.

1. are my nonces in the correct order?
2. how do my fees vary across my transactions?

In order to do that, I took the following steps:

Open my address in explorer
- Open web wallet and copy address
- Navigate to Stacks explorer
- Search for my address

Open each transaction in the explorer
- Open each tx in a new tab
- Review nonces starting with the account page + all txs
- Review fees starting with the first tx + all txs

Same for if I'm generating a new transaction (let's say I want to stack some STX following all that) - I have to go through the steps above to make sure my nonce is correct before submitting if I want to be extra sure. It'd be nice if there was a faster/easier way to do this.
